### DISCUSSION TITLE
Italy vs Europe performances

### DISCUSSION THREAD

**Discussion by AlessandroSalvetti** [Project-OSRM/osrm-backend#6840] [Q&A]
Hi everyone,
I've two separeted OSRM deployment, on two different servers. One only has Italy maps, the other the full Europe maps. The problem is that I see 15x difference in performances (runtime in seconds) when using the `route` service on the exact same coordinates.

I tested the `/route/v1/car` endpoint with the following parameters `alternatives=false steps=false overview=false continue_straight=false skip_waypoints=true generate_hints=false`

In the following image you can see the histogram of the time taken to respond. Each run of the experiment gets 50 random coordinates (in Italy) and send them both to ITA and EU server and runtime is recorded.

EU server is 15 times slower.
![image](https://github.com/Project-OSRM/osrm-backend/assets/69142481/4652ced6-e09f-4f8e-ab97-e80087d87b10)

For the deployment I'm using docker on both servers with 2CPU and 4Gb RAM (aws ec2 t3.medium). The real difference between the deployments is:
- ORSM ITA: the maps are fully loaded in memory
- OSRM EU: I had to use `--mmap=on` to make osrm running

Other paramenter used: `--max-matching-size 1000 --max-table-size 1000 --max-viaroute-size 1000 --algorithm mld`

What do you think is generating this difference?
Is there any point of attention I should take care of when working with EU maps?

Thanks for help

**Comment by datendelphin** [comment]
Italy seems to fit into RAM (you write fully loaded in memory) and Europe is for sure larger than the 4Gb of Ram you have. So routes will be answered from disk which is orders of magnitude slower than RAM.
You will need enough RAM to load the complete graph for your region, otherwise it gets terribly slow.

**Comment by AlessandroSalvetti** [comment]
Hi @datendelphin ,
you were right. 

I have increased the RAM of the EU server to 64Gb so that I do not need `--mmap=on` anymore, and the EU maps are fully loaded in memory.

Here are the results. 
![immagine](https://github.com/Project-OSRM/osrm-backend/assets/69142481/6ecc3b4d-bde0-4920-9c91-9bc29b3eebc9)

They are comparable, or rather, EU is even better (I suppose this is due to the highest number of cpu this server has, what do you think?)

Thanks so much for help!
